### PR TITLE
Allow "items" to be overridden on arrays using jsdoc annotations

### PR DESCRIPTION
**Pull request by dobesv**
I wanted to able to override the `items` of an array type using jsdoc annotations.  This PR makes that possible.

### CODE REVIEW

**Comment by dobesv** [comment]
Potentially fixes https://github.com/YousefED/typescript-json-schema/issues/236

File: README.md [hunk 1]
```diff
@@ -161,6 +161,65 @@ will be translated to
 
 Note that we needed to use `@TJS-type` instead of just `@type` because of an [issue with the typescript compiler](https://github.com/Microsoft/TypeScript/issues/13498).
 
+You can also override the type of array items, either listing each field in its own annotation or one
+annotation with the full JSON of the spec (for special cases).  This replaces the item types that would
```
Comment (domoritz) [inline README.md]:
```suggestion
annotation with the full JSON of the spec (for special cases). This replaces the item types that would
```

File: test/programs/annotation-items/main.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+interface MyObject {
+  /**
+   * @items {"type":"integer"}
+   */
+  a: number[];
```
Comment (domoritz) [inline test/programs/annotation-items/main.ts]: Why not use this pattern? 

```ts
{
  a: Integer[];
}

/**
 * @type integer
 */
type Integer = number;
```

Comment (dobesv) [reply test/programs/annotation-items/main.ts]: I haven't gotten that to work, for me it always comes through as just a plain number.

I couldn't find a test case showing that this is possible, and when I added one, the test failed to get the desired result.

Comment (dobesv) [reply test/programs/annotation-items/main.ts]: Has that worked for you?  I wonder if it's something that stops working in TypeScript 3.7

Comment (domoritz) [reply test/programs/annotation-items/main.ts]: It works in https://github.com/vega/ts-json-schema-generator. Anyway, I think it's the right way to support item typings so I'll close this pull request.

**Comment by dobesv** [comment]
I can't use https://github.com/vega/ts-json-schema-generator, it barfs with some parse error or something, and your proposed solution does not work.

What am I to do?

**Comment by dobesv** [comment]
Extremely disappointing 😭

**Comment by domoritz** [comment]
I'm sorry to disappoint you. I think the right thing would be to figure out how to support 

```ts
/**
   * @items {"type":"integer"}
   */
  a: number[];
```

**Comment by domoritz** [comment]
Hmm, thinking about this, this may actually not work since aliases are not fully supported here. I think you're right.

**Comment by domoritz** [comment]
I thought this was a hack to solve something where a better solution would be feasible but I don't think it is.

**Comment by dobesv** [comment]
Tests will not pass in TypeScript 3.7 for this PR until the other PR that fixes support for the `@TJS-` prefixed annotations is merged.

**Comment by dobesv** [comment]
Thank you for reconsidering!

**Comment by domoritz** [comment]
I'm human ;-)

**Pull request merged**